Sources/StreamChat/Utils/MulticastDelegate.swift

```diff
@@ -41,37 +41,37 @@ struct MulticastDelegate<T> {
     /// Sets the main delegate. If is nil, removes the main delegate.
     /// - Parameter mainDelegate: The main delegate.
     mutating func set(mainDelegate: T?) {
-        queue.async { [weak _mainDelegate] in
-            _mainDelegate?.removeAllObjects()
+        queue.sync {
+            _mainDelegate.removeAllObjects()
             
             if let delegate = mainDelegate {
-                _mainDelegate?.add(delegate as AnyObject)
+                _mainDelegate.add(delegate as AnyObject)
             }
         }
     }
 
     /// Replaces the current additional delegates with another collection of delegates.
     /// - Parameter additionalDelegates: The new additional delegates.
     mutating func set(additionalDelegates: [T]) {
-        queue.async { [weak _additionalDelegates] in
-            _additionalDelegates?.removeAllObjects()
-            additionalDelegates.forEach { _additionalDelegates?.add($0 as AnyObject) }
+        queue.sync {
+            _additionalDelegates.removeAllObjects()
+            additionalDelegates.forEach { _additionalDelegates.add($0 as AnyObject) }
```

Comment:
Especially this one could be concerning

Reply:
Since it just mutates the hash table then can't think of this being a real performance issue.

Reply:
My concern was if this could block the main thread somehow, since calling `sync` multiple times could block other stuff